python/lsst/pipe/tasks/postprocess.py

```diff
@@ -789,6 +791,181 @@ def _makeArgumentParser(cls):
+            Specifies if this is a Gen3 list of datarefs.
+
+        Returns
+        -------
+        visitSummary : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with per-detector summary information.
+        """
+        schema = afwTable.ExposureTable.makeMinimalSchema()
+        schema.addField('visit', type='I', doc='Visit number')
+        schema.addField('detector_id', type='I', doc='Detector number')
+        schema.addField('physical_filter', type='String', size=32, doc='Physical filter')
+        schema.addField('band', type='String', size=32, doc='Name of band')
+        schema.addField('psfSigma', type='F',
+                        doc='PSF model second-moments determinant radius (center of chip) (pixel)')
+        schema.addField('psfArea', type='F',
+                        doc='PSF model effective area (center of chip) (pixel**2)')
+        schema.addField('psfIxx', type='F',
+                        doc='PSF model Ixx (center of chip) (pixel**2)')
+        schema.addField('psfIyy', type='F',
+                        doc='PSF model Iyy (center of chip) (pixel**2)')
+        schema.addField('psfIxy', type='F',
+                        doc='PSF model Ixy (center of chip) (pixel**2)')
+        schema.addField('raCorners', type='ArrayD', size=4,
+                        doc='Right Ascension of bounding box corners (degrees)')
+        schema.addField('decCorners', type='ArrayD', size=4,
+                        doc='Declination of bounding box corners (degrees)')
+
+        cat = afwTable.ExposureCatalog(schema)
+        cat.resize(len(dataRefs))
+
+        cat['visit'] = visit
+
+        if not isGen3:
+            bbox = lsst.geom.BoxI(lsst.geom.PointI(0, 0), lsst.geom.PointI(1, 1))
+
+        for i, dataRef in enumerate(dataRefs):
+            if isGen3:
+                visitInfo = dataRef.get(component='visitInfo')
+                filter_ = dataRef.get(component='filter')
+                psf = dataRef.get(component='psf')
+                wcs = dataRef.get(component='wcs')
+                photoCalib = dataRef.get(component='photoCalib')
+                detector = dataRef.get(component='detector')
+                bbox = dataRef.get(component='bbox')
+            else:
+                # Note that we need to read the calexp because there is
+                # no magic access to the psf except through the exposure.
+                exp = dataRef.get(datasetType='calexp_sub', bbox=bbox)
+                visitInfo = exp.getInfo().getVisitInfo()
+                filter_ = exp.getFilter()
+                psf = exp.getPsf()
+                wcs = exp.getWcs()
+                photoCalib = exp.getPhotoCalib()
+                detector = exp.getDetector()
+                bbox = dataRef.get(datasetType='calexp_bbox')
+
+            rec = cat[i]
+            rec.setBBox(bbox)
+            rec.setVisitInfo(visitInfo)
+            rec.setWcs(wcs)
+            rec.setPhotoCalib(photoCalib)
```

Comment:
I'm not sure if this changed, but why aren't you persisting the `Detector` object here (I'll accept the argument that having the detector id trivially available in the outer metadata is useful)? There's a slot for it in `ExposureRecord`. I guess for now it's a constant thing, but eventually it has the possibility of changing during the survey.